contrib/mesos/pkg/scheduler/podtask/predicate.go

```diff
@@ -57,12 +61,18 @@ func NodeSelectorPredicate(t *T, offer *mesos.Offer) bool {
 
 	// check the NodeSelector
 	if len(t.Pod.Spec.NodeSelector) > 0 {
-		slaveLabels := map[string]string{}
-		for _, a := range offer.Attributes {
-			if a.GetType() == mesos.Value_TEXT {
-				slaveLabels[a.GetName()] = a.GetText().GetValue()
+		slaveLabels := node.SlaveAttributesToLabels(offer.Attributes)
+
+		// add non-mesos labels
```

Comment:
I might do this the other way around: populate `slaveLabels` from the node, and them update them with the results of SlaveAttributesToLabels: the slave attributes always take priority then. You could probably get rid of the reference to LabelPrefix here then (and then make it package-private).